servers/src/test/java/tachyon/master/next/IndexedSetTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.next;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Assert;
+
+public class IndexedSetTest {
+  private static class Pair {
+    // mInt is private, mDouble is public, this is deliberate to assure both private and public
+    // fields can be accessed.
+    private int mInt;
+    public double mDouble;
```

Comment:
Not sure it is a good practice to have the second field of `double` in this class for test, as the equality condition on `double` is very tricky and sometimes not perfectly precise.